extensions/azcli/src/az.ts

```diff
@@ -285,36 +267,70 @@ async function executeAzCommand(command: string, args: string[], additionalEnvVa
+			response = await vscode.window.showErrorMessage(loc.noAzureCLI, ...responses);
+			if (response === loc.doNotAskAgain) {
+				await setConfig(azCliInstallKey, AzDeployOption.dontPrompt);
+			}
+		}
+		throw err;
+	}
+
 	const versionOutput = await executeAzCommand(`"${path}"`, ['--version']);
-	const version = parseArcExtensionVersion(versionOutput.stdout);
-	const semVersion = new SemVer(version);
-	//let response: string | undefined;
+	const azVersion = parseVersion(versionOutput.stdout);
+	const arcVersion = parseArcExtensionVersion(versionOutput.stdout);
 
-	if (LATEST_AZ_ARC_EXTENSION_VERSION.compare(semVersion) === 1) {
-		// If there is a greater version of az arc extension available, prompt to update
-		// Commented the install/update prompts out until DoNotAskAgain is implemented
-		// const responses = [loc.askLater, loc.doNotAskAgain];
-		// response = await vscode.window.showInformationMessage(loc.requiredArcDataVersionNotAvailable(latestAzArcExtensionVersion, version), ...responses);
-		// if (response === loc.doNotAskAgain) {
-		// 	await setConfig(azRequiredUpdateKey, AzDeployOption.dontPrompt);
-		// }
-	} else if (LATEST_AZ_ARC_EXTENSION_VERSION.compare(semVersion) === -1) {
-		// Current version should not be greater than latest version
-		// Commented the install/update prompts out until DoNotAskAgain is implemented
-		// vscode.window.showErrorMessage(loc.unsupportedArcDataVersion(latestAzArcExtensionVersion, version));
+	if (arcVersion === undefined) {
+		// If no arcdata was found, prompt to install
+		const arcInstallNeededConfig = <AzDeployOption>getConfig(azArcdataInstallKey);
+		if (arcInstallNeededConfig !== AzDeployOption.dontPrompt) {
+			response = await vscode.window.showInformationMessage(loc.arcdataExtensionNotInstalled, ...responses);
+			if (response === loc.doNotAskAgain) {
+				await setConfig(azArcdataInstallKey, AzDeployOption.dontPrompt);
+			}
+		}
+	} else {
+		// If arcdata was found, check if it's up to date
+		const semVersion = new SemVer(<string>arcVersion);
+		const arcUpdateNeededConfig = <AzDeployOption>getConfig(azArcdataUpdateKey);
+
+		if (LATEST_AZ_ARC_EXTENSION_VERSION.compare(semVersion) === 1 && arcUpdateNeededConfig !== AzDeployOption.dontPrompt) {
+			// If there is a greater version of arcdata extension available, prompt to update
+			response = await vscode.window.showInformationMessage(loc.requiredArcDataVersionNotAvailable(latestAzArcExtensionVersion, <string>arcVersion), ...responses);
+			if (response === loc.doNotAskAgain) {
+				await setConfig(azArcdataUpdateKey, AzDeployOption.dontPrompt);
+			}
+		} else if (LATEST_AZ_ARC_EXTENSION_VERSION.compare(semVersion) === -1) {
+			// Current version should not be greater than latest version
+			vscode.window.showErrorMessage(loc.unsupportedArcDataVersion(latestAzArcExtensionVersion, <string>arcVersion));
+		}
 	}
-	return new AzTool(path, version);
+
+	return new AzTool(path, <string>azVersion);
```

Comment:
I'd consider storing either just the arcdata version here or both versions - just having the az CLI version doesn't seem super useful at the moment

Reply:
Got it, if it's cool with you that's something I will include in the next release which will be likely be soon after this upcoming one on 8/3 morning.

Reply:
Pushed a change to this PR branch that adds both az version and arcdata version to the AzTool (the change also includes changing parseVersion to regex).